---
title: "Using FREEC to analyse CNVs in wild maize"
author: "Simon Renny-Byfield"
date: "February 3, 2015"
output: html_document
---

As part of my project we are interested in calling PAV-CNV in 20 teosinte individuals from a single population near Palmar Chico, Mexico. I have tried a number of options, none of which are producing what seems like sensible results and so I have decided to pursue different approaches.

[FREEC]("http://bioinformatics.oxfordjournals.org/content/27/2/268.long") infers "control-free copy number alteration (CNA)" detection using deep-sequencing data. The installation is standard and won't be detailed here but there are quite a few auxiliary files that are needed and I will go through how these are generated.

##The config file##

FREEC requires the generation of a config file. An example of such a file is given below:

```
[general]

chrLenFile = hs18_chr.len
coefficientOfVariation = 0.062
ploidy = 3

#breakPointThreshold = -.002;
#window = 50000
#chrFiles = hg18/hg18_per_chromosome
#outputDir = test
#degree=3
#intercept = 0

[sample]

mateFile = HCC1143.arachne
inputFormat = arachne
mateOrientation = 0

#mateCopyNumberFile = HCC1143.arachne_sample.cpn

[control]

mateFile = HCC1143_BL.arachne
inputFormat = arachne
mateOrientation = 0

#mateCopyNumberFile = HCC1143_BL.arachne_control.cpn
```

also see [here]("http://bioinfo-out.curie.fr/projects/freec/tutorial.html#CONFIG") for detailed descriptions of what files are needed. From this point on I am using this webpage as guidance for generating the required files. I already have the cHr length file and it looks like this:
```
1  1	301476924
2	2	237917468
3	3	232245527
4	4	242062272
5	5	217959525
6	6	169407836
7	7	176826311
8	8	175377492
9	9	157038028
10	10	149632204
```
This is for maize `AGPv3.22`.

So the next step is to generate single sequences for each chromosome using a very specific genome references sequence. In this case it is version 3.22, a version of this can be found in `/home/vince251/data/refgen3-hapmap/maize3.fa.gz` which is the exact sequence that the `.bam` files were mapped to. So now to split the sequence up...

The sequences have been split using the following BioPerl heavy script:
```
#!usr/bin/perl
use strict;
use warnings;
use Bio::SeqIO;

###############################
#Simon Renny-Byfield, UC Davis, 7th November 2014, version 1
###############################

#usage script.pl <fasta> 

#This script takes fasta files and splits the sequences into individual files

#Initialize the Bio:SeqIO object; 
my $seqIO = Bio::SeqIO -> new ( -format => "fasta" , -file => $ARGV[0] );

#also intilize a file to contain all the name
#start looping through the sequences
while ( my $seqObj = $seqIO->next_seq() ) {
  #get the sequence id
	my $name = $seqObj->id;
	#open up a new file with the seq id as name
	open(NEWFASTA, ">$name".".fasta" ) || die "could not open file:$!\n";
	#print the name of the sequence
	print NEWFASTA ">$name\n";
	#grab the sequence string
	my $seqstr = $seqObj->seq();
	#print it
	print NEWFASTA "$seqstr\n";
	#close the file handle
	close(NEWFASTA) 
}#while

exit;
```

I will set `maxThreads1=10` in the config file.

The final `config` file looks like this:
```
[general]

chrLenFile = /group/jrigrp4/freec/chrLenFile.txt
coefficientOfVariation = 0.062
ploidy = 2
chrFiles=/group/jrigrp4/freec/maize3/
maxThreads=10
sex=XX

#breakPointThreshold = -.002;
window = 1000
#degree=3
#intercept = 0

[sample]

mateFile = /group/jrigrp4/cn.mops/TIL_data/TIL01_sorted.bam
inputFormat = BAM
mateOrientation = FR

#mateCopyNumberFile = HCC1143.arachne_sample.cpn

[control]

#mateFile = HCC1143_BL.arachne
#inputFormat = BAM
#mateOrientation = FR

#mateCopyNumberFile = HCC1143_BL.arachne_control.cpn
```

The above template is the nused to generate a config script, one for each of the bam files you wish to analyze. This acieved with a very simple perl script detialed below:
```
#!usr/bin/perl
use strict;
use warnings;

# usage script.pl /path/to/the/bam/files
###
# Read in the .bam files
###
opendir my $dir, "$ARGV[0]" or die "Cannot open directory: $!";
my @files = readdir $dir;
closedir $dir;

@files = grep ( /.bam$/ , @files);

###
# now create a freec config script for each bam file.
###
foreach ( @files ) {
  my $dir ="/group/jrigrp4/freec/output/".$_;
	mkdir $dir;
	open(OUT , ">/group/jrigrp4/freec/config/config_$_.conf" ) || die;
	print OUT "[general]
	
chrLenFile = /group/jrigrp4/freec/chrLenFile.txt
coefficientOfVariation = 0.062
ploidy = 2
chrFiles=/group/jrigrp4/freec/maize3/
maxThreads=10
sex=XX
outputDir=/group/jrigrp4/freec/output/$_
#breakPointThreshold = -.002;
window = 1000
#degree=3
#intercept = 0

[sample]

mateFile = /group/jrigrp4/cn.mops/data/$_
inputFormat = BAM
mateOrientation = FR

#mateCopyNumberFile = HCC1143.arachne_sample.cpn

[control]

#mateFile = HCC1143_BL.arachne
#inputFormat = BAM
#mateOrientation = FR

#mateCopyNumberFile = HCC1143_BL.arachne_control.cpn";
	close OUT;

} #foreach

exit;
```

This worked just fine, and I now have a bunch of config scripts that I can submit as a slurm array job, with a typical slurm script such as:

```
#!/bin/bash
#OUTDIR=/group/jrigrp4/freec/config
#SBATCH -D /group/jrigrp4/freec/config
#SBATCH -o /group/jrigrp4/freec/logs/freec_out_log-%j.txt
#SBATCH -e /group/jrigrp4/freec/logs/freec_err_log-%j.txt
#SBATCH -J freec
#SBATCH --mem=12000
#SBATCH --cpus-per-task 10
#SBATCH --array=0-20

files=(*.conf)

#cmd="samtools index ${files[$SLURM_ARRAY_TASK_ID]}"

echo "Starting freec job: "
date

cmd="freec -conf ${files[$SLURM_ARRAY_TASK_ID]}"
echo $cmd
eval $cmd

echo "Job Ending: "
date
```
where each config file ends in `.conf` and is stored in `/group/jrigrp4/freec/config`. This job is currently running on farm.




